components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/store/HumanTaskStore.java

```diff
@@ -438,10 +440,21 @@ public HumanTaskDeploymentUnit createNewDeploymentUnit(File humanTaskFile,
                                                            int tenantId,
                                                            long version,
                                                            String md5sum) throws HumanTaskDeploymentException {
-        ArchiveBasedHumanTaskDeploymentUnitBuilder builder =
-                new ArchiveBasedHumanTaskDeploymentUnitBuilder(humanTaskFile, tenantId, version, md5sum);
-        HumanTaskDeploymentUnit newHumanTaskDeploymentUnit = builder.createNewHumanTaskDeploymentUnit();
-        return newHumanTaskDeploymentUnit;
+        try {
+            ArchiveBasedHumanTaskDeploymentUnitBuilder builder = new ArchiveBasedHumanTaskDeploymentUnitBuilder(
+                    humanTaskFile, tenantId, version, md5sum);
+            HumanTaskDeploymentUnit newHumanTaskDeploymentUnit = builder.createNewHumanTaskDeploymentUnit();
+            return newHumanTaskDeploymentUnit;
+        } catch (HumanTaskDeploymentException deploymentException) {
+            if (log.isDebugEnabled()) {
+                log.debug("humanTask:" + humanTaskFile.getName()
+                        + " deployment failed, removing the extracted human task directory.");
+            }
```

Comment:
The debug log is there to mention that we are removing the extracted directory. Anyway an exception is thrown. So the error log will be logged later. (see the next few lines in the full code view). So I guess this is OK.